### PR TITLE
Summoner/validation form logic

### PR DESCRIPTION
**Pull request by jordanlesich**
## GitHub Issue

https://github.com/HausDAO/daohaus-monorepo/issues/60

## Changes

- Builds in internal logic for basic form validation
- Utilises react-hook-form's native validation system (was able to remove my own hacky system). 
- Builds form segment logic (changing UI based on formvalues)
- Handles field data modifiers (ex. parsing address, shares, loot from string and sending that data to formValues)
- Handles complex validations that can read post-modification data 
- Time Picker Input Select
- Many Typeguard and validation utils. 

Note: 

Did not end up using yup. While react-hook-form does have some decent integrations + docs on yup, it seems that yup likes a validation scheme at the top level of the form. The potential challenge I wanted to avoid was dealing with top-level validations, trying to time it with RHF validation timing and the custom modifiers we use. 

In the end, I decided to deal with the tedium of making some typecheck fns (or copying them over from V2) over dealing integration issues. I'm still open to reconsidering yup, but it would need to do the following: 

- validate at the field level and allow for components to have their own default validations/
- Handle custom value 
- Somehow work easily with the formBuilder API (would we pass in the schema as props?) when it gets built
- Allow for custom error messages (I'm sure it does that) 
- Allow for custom validations (I'm sure it does that as well; just making sure)

## Packages Added

"@gnosis.pm/safe-contracts" (will need for TX)

## Checks

Before making your PR, please check the following:

- [x] Critical lint errors are resolved
- [x] App runs locally
- [x] App builds locally (run the build command for *any impacted package* and check for any errors before the PR)

### CODE REVIEW

File: apps/summon-app/src/layouts/ShamanSegment.tsx [hunk 1]
```diff
@@ -1,22 +1,54 @@
...
       title="Starting Shamans"
       description="Shamans are very powerful as they can have administrative control over voting and non-voting stakes. Be very careful adding shamans. "
       formArea={
         <TextAreaSection>
           <TemporaryLink className="link">How to add a Shaman</TemporaryLink>
-          <ParSm className="number-display">0 Shamans</ParSm>
+          <ParSm className="number-display">{amtShamans} Shamans</ParSm>
           <WrappedTextArea
             label="Addresses & Permissions"
             placeholder="0x00000000000000000000000000 3"
```
Comment (skuhlmann) [inline apps/summon-app/src/layouts/ShamanSegment.tsx:43]: ZERO_ADDRESS might make a nice reusable constant.

File: apps/summon-app/src/utils/common.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    'Shares are required and must be a number. Check formatting rules in tooltip above',
+  ADDRESS_ERR:
+    'Member addresses are required and must be valid Ethereum Addresses. Check formatting rules in tooltip above',
+  SHAMAN_ADDR_ERR:
+    'Shaman addresses are required and must valid ethereum addresses',
+  PERMISSION_ERR: 'Shaman permissions are required and must be a number',
+};
+
+////////////////////Members Segment////////////////////////
+export const validateMemberData = (
```
Comment (skuhlmann) [inline apps/summon-app/src/utils/common.ts:19]: :+1:

File: apps/summon-app/src/utils/summonModel.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+// );
+// const METADATA_CONFIG = {
+//   CONTRACT: LOCAL_ABI.BAAL,
+//   ACTION: 'executeAsBaal',
+//   ARGS: {
+//     TO: CONTRACT_ADDRESSES.POSTER['0x4'],
+//     VALUE: 0,
+//     DATA: METADATA,
+//   },
+// };
```
Comment (alexkeating) [inline apps/summon-app/src/utils/summonModel.ts:67]: Should we remove these comments?

File: libs/ui/src/components/atoms/Button/Button.tsx [hunk 4]
```diff
@@ -17,6 +17,7 @@ export type ButtonProps = {
   onClick?: React.MouseEventHandler<HTMLButtonElement>;
   IconLeft?: IconType;
   IconRight?: IconType;
+  type?: 'button' | 'submit' | 'reset' | undefined;
```
Comment (alexkeating) [inline libs/ui/src/components/atoms/Button/Button.tsx:20]: Isn't undefined handled by the ?

**Review by alexkeating — APPROVED** [review]

File: apps/summon-app/src/utils/summonModel.ts (hunk 3 above)
Comment (jordanlesich) [reply apps/summon-app/src/utils/summonModel.ts:67]: It's there for when I put in the TX

File: apps/summon-app/src/layouts/ShamanSegment.tsx (hunk 1 above)
Comment (jordanlesich) [reply apps/summon-app/src/layouts/ShamanSegment.tsx:43]: I'll add to commons

**Pull request merged**